iothub/device/devdoc/API Design.md

```diff
@@ -0,0 +1,302 @@
+public Task SubscribeToWritablePropertyEvent(Action<Properties, object> callback, string componentName = default, object userContext = default, CancellationToken cancellationToken = default);
+```
+
+<details>
+<summary>Properties</summary>
+
+```csharp
+/// <summary>
+/// A container for properties for your device
+/// </summary>
+public class Properties
+{
+    private PropertyCollection _readOnlyProperties;
+    /// <summary>
+    /// Initializes a new instance of <see cref="Properties"/>
+    /// </summary>
+    public Properties()
+    {
+        Writable = new PropertyCollection();
+        _readOnlyProperties = new PropertyCollection();
+    }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="Properties"/> with the specified collections
+    /// </summary>
+    /// <param name="writablePropertyCollection">A collection of writable properties returned from IoT Hub</param>
+    /// <param name="readOnlyPropertyCollection">A collection of read-only properties returned from IoT Hub</param>
+    public Properties(PropertyCollection writablePropertyCollection, PropertyCollection readOnlyPropertyCollection)
+    {
+        Writable = writablePropertyCollection;
+        _readOnlyProperties = readOnlyPropertyCollection;
+    }
+
+    /// <summary>
+    /// Gets and sets the writable properties.
+    /// </summary>
+    [JsonProperty(PropertyName = "desired", DefaultValueHandling = DefaultValueHandling.Ignore)]
+    public PropertyCollection Writable { get; private set; }
+
+    /// <summary>
+    /// Get the property from the propeties collection
+    /// </summary>
+    /// <param name="propertyName"></param>
+    /// <returns></returns>
+    public dynamic this[string propertyName]
+    {
+        get
+        {
+            return _readOnlyProperties[propertyName];
+        }
+    }
+
+    /// <summary>
+    /// Converts a <see cref="TwinProperties"/> collection to a properties collection
+    /// </summary>
+    /// <param name="twinProperties">The TwinProperties object to convert</param>
+    /// <returns></returns>
+    public static Properties FromTwinProperties(TwinProperties twinProperties)
+    {
+        if (twinProperties == null)
+        {
+            throw new ArgumentNullException(nameof(twinProperties));
+        }
+
+        return new Properties()
+        {
+            _readOnlyProperties = (PropertyCollection)twinProperties.Reported,
+            Writable = (PropertyCollection)twinProperties.Desired
+        };
+    }
+}
+```
+</details>
+
+<details>
+<summary>PropertyCollection</summary>
+
+```csharp
+/// <summary>
+/// A collection of properties for the device
+/// </summary>
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1010:Generic interface should also be implemented", Justification = "<Pending>")]
+public class PropertyCollection : TwinCollection
+{
+}
+```
+</details>
+
+
+<details>
+<summary>WritableProperty</summary>
```

Comment:
This isn't referenced anywhere - is it vestigial?

Reply:
this was supposed to be "WritablePropertyResponse", renamed.